brave-intl/bat-go
# Update SAM Development Github Actions YAML

- starting on an email serverless service
- adding unsubscribe capabilities
- add unsubscribe reference to the template data
- fixing up go struture issues
- updates to go mod tidy all modules
- update all go mod deps, test each module seperately
- linting and fixing ci lint checks
- name each step for linting different mods
- make different jobs for linting mods
- errcheck lint issue correction
- adding linting of serverless modules
- move client creation to global
- update make lint to match what ci linting does, update ci linting
- fix dockerfile
- fix dockerfile
- more dockerfile
- fix schema location for test
- fix rewards parameters to match new schema
- check value of prometheus.Register
- adding auth check
- adding configuration set for send email
- use secrets manager for the values
- Run go mod tidy in webhook to pull in recent deps
- typos
- correct auth token behavour for email webhook
- input payload validation
- add security_opts to docker compose
- fix questionable assignments
- move binary from src/main/ from builder
- Added SSM Parameter Store Variables to SAM Template; Updated Github Actions with IAM information and S3 bucket artifacts
- Updated sam-dev.yml
- updated sam-dev.yml

### Summary



### Type of change ( select one )

- [ ] Product feature



### Tested Environments


### Before submitting this PR:

- [ ] Does your code build cleanly without any errors or warnings?
- [ ] Have you used auto closing keywords?
- [ ] Have you added tests for new functionality?
- [ ] Have validated query efficiency for new database queries?
- [ ] Have documented new functionality in README or in comments?
- [ ] Have you squashed all intermediate commits?
- [ ] Is there a clear title that explains what the PR does?
- [ ] Have you used intuitive function, variable and other naming?
- [ ] Have you requested security / privacy review if needed
- [ ] Have you performed a self review of this PR?

### Manual Test Plan:
